koala/ast/graph.py

```diff
@@ -182,21 +178,28 @@ def eval_range(rng, rng2=None):
                     rng2 = rng2.split('!')[1]
                 # return eval_range('%s:%s' % (rng, rng2))
                 return self.evaluate_range(CellRange('%s:%s' % (rng, rng2),sheet), False)
-                
+        
         try:
-            # print "Evalling: %s, %s" % (cell.address(),cell.python_expression)
+            print "Evalling: %s, %s" % (cell.address(),cell.python_expression)
             vv = eval(cell.compiled_expression)
-            #print "Cell %s evalled to %s" % (cell.address(),vv)
+
             if vv is None:
                 print "WARNING %s is None" % (cell.address())
+            # elif isinstance(vv, (List, list)):
+            #     print 'Output is list => converting', cell.index
+            #     vv = vv[cell.index]
             cell.value = vv
         except Exception as e:
             if e.message.startswith("Problem evalling"):
                 raise e
             else:
                 raise Exception("Problem evalling: %s for %s, %s" % (e,cell.address(),cell.python_expression)) 
-        
-        return cell.value
+
+        try:
+            return cell.value
+        except:
+            for f in missing_functions:
+                print 'MISSING', f
```

Comment:
The primary idea was to collect all missing functions once and for all, to get a list of all functions that we need to work on. This was a try, never really worked, i'm sure this is actually possible.